chia/wallet/wallet_state_manager.py

```diff
@@ -634,24 +634,25 @@ async def determine_coin_type(
         if coin_spend is None:
             return None, None
 
+        puzzle = Program.from_bytes(bytes(coin_spend.puzzle_reveal))
+
         # Check if the coin is a CAT
-        cat_matched, cat_curried_args = match_cat_puzzle(Program.from_bytes(bytes(coin_spend.puzzle_reveal)))
+        cat_matched, cat_curried_args = match_cat_puzzle(puzzle)
         if cat_matched:
             return await self.handle_cat(cat_curried_args, parent_coin_state, coin_state, coin_spend)
 
         # Check if the coin is a NFT
         #                                                        hint
         # First spend where 1 mojo coin -> Singleton launcher -> NFT -> NFT
         try:
-            uncurried_nft = UncurriedNFT.uncurry(Program.from_bytes(bytes(coin_spend.puzzle_reveal)))
+            uncurried_nft = UncurriedNFT.uncurry(puzzle)
+            return await self.handle_nft(coin_spend, uncurried_nft)
         except Exception:
             # This is not a NFT coin, skip NFT handling
             pass
-        else:
-            return await self.handle_nft(coin_spend, uncurried_nft)
```

Comment:
I moved this up into the `try`-block, as I think it simplifies the code (and saves one line)

Reply:
yep, else is confusing, it's not executed in exceptions, but it's not clear